### ISSUE TITLE
Search fixes

### ISSUE THREAD

**Issue by jmartinm** [inspirehep/inspire-next#716]
Here we can keep track of searches that are not working (or not working as expected).
- [x] `find a` only works with author full name. For example `find a Maschke` will produce 0 results, while `find a Maschke, a.W.` will work. In the production website, writing the surname of the author seems to be enough to produce results.
- [x] Searching for an author's surname produces 0 results.
- [ ] Searching for an e-Print produces wrong results. A search for `hep-ph/9607327` will display as a result the record with e-Print `hep-ph/9610500`, despite the fact that there is another record with exactly the search query we're looking for (recid: 420758)
- [ ] DOI should match first. With inspire-next demo records, a query for `10.1088/1748-0221/7/10/P10002` gives the record with that DOI as a second result. The first one seems quite unrelated.
- [ ] Journal search should be part of google-like and boosted. A search for [JINST 6](https://inspirehep.net/search?ln=en&ln=en&p=JINST+6&of=hb&action_search=Search&sf=earliestdate&so=d&rm=&rg=25&sc=0) gives top results on INSPIRE related to the journal name while it does not on inspire-next. Problem is that `publication_info` is not added to the fields we look into in google syntax.

First three are extracted from https://github.com/inspirehep/inspire-next/issues/581#issuecomment-163668728

**Comment by Panos512** [comment]
inspirehep/inspire-next#715 fixes the first two issues

**Comment by jmartinm** [comment]
The third one seems to still be a problem even after merging https://github.com/inspirehep/invenio-search/commit/f2c9a074cc804e745f1f959fa7d73199be3774da

I have loaded record https://inspirehep.net/record/420758/export/xm into my system and searching for:

`hep-ph/9607327`

does not bring show that record as a result.

It seems a bit weird as the query is interpreted correctly:

```
{'multi_match': {'message': {'operator': 'or', 'query': u'hep-ph/9607327 ', 'fields': ['title^3', 'title.raw^10', 'abstract^2', 'abstract.raw^4', 'author^10', 'author.raw^15', 'reportnumber^10', 'eprint^10', 'doi^10'], 'zero_terms_query': 'all'}}}
```

And the field `eprint` is defined as not analyzed https://github.com/inspirehep/inspire-next/blob/master/inspirehep/base/searchext/mappings/hep.json#L154

**Comment by Panos512** [comment]
Looks like the problem is `eprint` and `doi` are not analyzed. Is there a reason we need them non-analyzed on both the original field and the field we copy them to?

**Comment by kaplun** [comment]
DOI/eperint are identifiers, so they make sense only as a whole string, not as a tokenized/transformed one.

**Comment by Panos512** [comment]
Yes but we keep them in two fields (e.g dois.value and doi) we could analyze the one to try and match the google-like search and keep the other one unanalyzed for specific searches

**Comment by kaplun** [comment]
@Panos512 I am trying out an overall clean-up job of `hep.json`: see my current effort in https://github.com/kaplun/inspire-next/commit/1ab1a046272a243a359fd49a7b1c740c7e5347ef

In order to implement this I have first:
* Sorted and prettified the current JSON (in https://github.com/kaplun/inspire-next/commit/e57c903a23bbc59b45a360d9c5c563edd7fbaf81)
* Dumped from production what ES has discovered so far of our records (through dynamic mapping)
* Transformed hep-0.0.1.json JSONSchema into a basic ES mapping (using inveniosoftware/es-jsonschema)
* Merged the three file resulting
* Adding lots of stuff...

Maybe we can continue from that?

**Comment by Panos512** [comment]
Looks like the problem (for bullets 3,4) was a tailing whitespace at the google-like search queries. Since the fields are not-analyzed elasticsearch was looking for "exact matches" and the whitespace was preventing it from matching the correct results.

I am now uploading a fix.

**Comment by kaplun** [comment]
@Panos512 can you summarize which issues you are currently tackling from this list?

**Comment by StellaCh** [comment]
Addressed by https://github.com/inspirehep/inspire-query-parser/

**Comment by kaplun** [comment]
So we are sure the above 4 points have been fixed?

**Comment by StellaCh** [comment]
4/5 are fixed and the journal search is one of the future steps that we have documented.
